File: MinimumSizeSubarraySumGreaterOrEqualsTargetActualArray.py
def maxSumOfArray(nums , k):
    maxSum = float('-inf')
    countSum = 0
    lower = 0
    
    for i in range(len(nums)):
        countSum += nums[i]
        if i >= k - 1 :
            if countSum > maxSum:
                maxSum = countSum
                maxSubArray = nums[i -k + 1: i + 1]
            countSum -= nums[i - k + 1]
                 
            
    return maxSum , maxSubArray 
        
if __name__ == "__main__":
    nums = [2, 1, 5, 1, 3, 2]
    k = 3
    result, subarray = maxSumOfArray(nums, k)
    print("Maximum sum of subarray of size", k, "is:", result)
    print("Maximum sum of subarray of size", k, "is:", subarray)

File: MinimumSizeSubarraySumGreaterOrEqualsTargetActualArray.java

import java.util.*;
class Main {
    public static void main(String[] args) {
        int [] nums = {2,3,1,2,4,3};
        int k = 3;
         System.out.println(Arrays.toString(MinimumSizeSubarraySumGreaterOrEqualsTarget(nums,k)));
    }
    
    public static int [] MinimumSizeSubarraySumGreaterOrEqualsTarget(int [] nums , int k){
        
        int maxSum = Integer.MIN_VALUE;
        int startIndex = 0;
        int sumCount = 0;
        
        for(int i = 0; i < nums.length; i ++){
            sumCount += nums[i];
            
            if (i >= k -1){
                if (sumCount > maxSum){
                    maxSum = sumCount;
                    startIndex = i - k + 1;
                }
            }
        }
          int [] maxSubArray = Arrays.copyOfRange(nums,startIndex, startIndex + k);
            return maxSubArray;
     
    }

}

// this function returns maxmum SubArray of an array but the actual sub array not the sum 
